### PR TITLE
[@types/react] Deprecate `ReactComponentElement`

### PR DESCRIPTION
**Pull request by ghost**
Closes #68457 

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test react`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by eps1lon** [comment]
Do we know what this was used for and what should be used instead? Is there any usage of this type in this repo?

**Comment by codershiba** [comment]
> Do we know what this was used for and what should be used instead? Is there any usage of this type in this repo?

https://github.com/search?q=repo:DefinitelyTyped/DefinitelyTyped%20ReactComponentElement&type=code
Seems like only `brainhubeu__react-carousel` was using this interface. Other than that `babel-plugin-react-html-attrs` had the same interface redeclared in its types

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/react/v16/index.d.ts
1. types/react/v17/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 459bdf7651a0cc5ab96ae4926c99544cffb4e25e
</p>

File: types/react/v17/index.d.ts [hunk 1]
```diff
@@ -113,6 +113,9 @@ declare namespace React {
         key: string | null;
     }
 
+    /**
+     * @deprecated
```
Comment (eps1lon) [inline types/react/v17/index.d.ts:117]: We don't need to deprecate it for the old types since we're not going to remove it from them.

**Review by eps1lon — CHANGES_REQUESTED** [review]

**Comment by eps1lon** [comment]
> Seems like only brainhubeu__react-carousel was using this interface.

And how would we resolve the breaking change there? This is essential to deprecating a type. We need to have an answer for the migration story.

**Comment by codershiba** [comment]
> > Seems like only brainhubeu__react-carousel was using this interface.
> 
> And how would we resolve the breaking change there? This is essential to deprecating a type. We need to have an answer for the migration story.

How about redeclaring the types in the package rather than importing it from react like `babel-plugin-react-html-attrs` does?

**Comment by codershiba** [comment]
> > Seems like only brainhubeu__react-carousel was using this interface.
> 
> And how would we resolve the breaking change there? This is essential to deprecating a type. We need to have an answer for the migration story.

How about redeclaring the types in the package rather than importing it from react like `babel-plugin-react-html-attrs` does?

**Comment by eps1lon** [comment]
Yeah that makes sense if that's the best migration path we can offer. But it needs to be documented in the PR.

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/brainhubeu__react-carousel/index.d.ts
1. types/brainhubeu__react-carousel/brainhubeu__react-carousel-tests.tsx

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2981e2f48bfaf0d8d680c541a91de6ed74ceac36
</p>

**Comment by rbuckton** [comment]
@eps1lon do the latest changes satisfy your review feedback?

**Review by eps1lon — APPROVED** [review]
Makes sense to deprecate. I generally try to keep the utility types to a minimum to make learning React easier. This seems like a good candidate for removal considering it's only used in one other DT package.

**Review by jakebailey — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]

**Comment by codershiba** [comment]
Ready to merge

**Pull request merged**